### DISCUSSION TITLE
2024-08-12 월요일

### DISCUSSION THREAD

**Discussion by eunohhh** [nbc-final-a10/TripBuddy#252] [기록보드]
2024-08-12 월요일 기록보드입니다.

## 여행가고십조(A10) 일일 공통 일과표

| 시간 | 일과 |  기록 |
| :-: | :-: | :-: |
| 09:00 ~ 10:00 | 아침 회의 | |
| 10:00 ~ 11:00 | 최종 주간 발제(zoom meeting) | |
| 11:00 ~ 13:00 | 집중 개발 time | |
| 13:00 ~ 14:00 | 점심 식사 | |
| 14:00 ~ 18:00 | 집중 개발 time | |
| 18:00 ~ 19:00 | 저녁 식사 | |
| 19:00 ~ 20:00 | 저녁 회의 | |
| 20:00 ~ 21:00 | 5분 기록보드 작성 / 개인 회고 | |

```
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- 내용

### 트러블 슈팅

- 내용

### 내일 계획

- 내용
```

```
// 이미지 테이블
<table>
  <tr>
    <th>이미지 1번 제목</th>
    <th>이미지 2번 제목</th>
  </tr>
  <tr>
    <td><img src="이미지 주소" width="300" height="300"/></td>
    <td><img src="이미지 주소" width="300" height="300"/></td>
  </tr>
</table>
```

**Comment by eunohhh** [comment]
## 개인 기록보드 서식 (오은)

### 오늘 한 일

- [x] : 여정 카드에 플필사진 4개까지 보이게끔
- [x] : 여정 찜하기 
- [ ] : 글수정 페이지 만들기
- [ ] : 얼러트 모달 수정
- [x] : 비번찾기 뒤로가기 동작 수정
- [ ] : 상대플필보기에서 여정보기 시에도 화면 지진현상
- [ ] : 회원가입 디바운싱 ??
- [ ] : 여정 임시 저장 ??
- [ ] : ai 이미지
- [ ] : 글쓰기 페이지 레이아웃 검토
- [ ] : 오늘날짜를 여정 시작일로 할 경우 뜨는 에러 처리
- [ ] : 컨디셔널 라우트
- [ ] : 온보딩으로 보내기처리

### 처리 된 사항
- [x] : 스토리 글쓰기 시 옵션 조금 상세하게 
- [x] : 튜토리얼 데스크탑에서 헤더 숨기기 
- [x] : pwa 연구 - 하단 참조(ios에서는 불가능)
- [x] : 스토리 삭제 
- [x] : 스토리 좋아요 
- [x] : 스토리 좋아요 옵티미스틱 
- [x] : 모바일에서 스토리 인풋문제 
- [x] : 홈화면 카드에서도 스토리 좋아요 노출하기
- [x] : 네이버 검수요청 
- [x] : 미들웨어 설정
- [x] : 여정 데스크탑 페이지네이션 
- [x] : 여정 관련 UI 수정작업
- [x] : 여정 모바일 무한스크롤
- [x] : 스토리 데탑/모바일 위치 맞추기
- [x] : 데스크탑 홈에서 마우스 스크롤 동작 수정 - 세로스크롤 가능하게하고, 화살표로 목록 이동하고, 좌우스크롤은 옵션
- [x] : error.js not-found.js 등 없는 기능들 
- [x] : 홈배너 로직 수정 - 탠스택쿼리
- [x] : 메인 스테일타임 점검 - 5분으로 설정
- [x] : 하단바 svgr 및 fixed
- [x] : 프로필 에디트 라우트 이동
- [x] : 2. (버그) 여정 상세보기, 버디 보기가 동작하지 않음 - 재현 불가
- [x] : 5. (제안) 스토리 자동으로 넘어가도록 - 미수용 : 스토리 자동 넘어가기는 구현하지 않을 생각이었음
- [x] : 6. (버그) 소개글 수정후 바로 적용 안됨 - 재현 불가
- [x] : 8. (버그) 소개글 수정후 레이아웃 바뀜 - 재현 불가
- [x] : 9. (버그) 홈페이지 스크롤 시 좌우 지진 현상 발생 - ***수정 완료*** 데스크탑과 모바일 나눠서 8/9 저녁회의때 보여드린대로 바르게 동작하게 수정함
- [x] : 10. (제안) 인기 스토리를 -> 인기 버디즈의 스토리로 변경 - 미수용 : 별 차이 없음
- [x] : 11. (제안) 모집 중 여정에서 관련 숙박시설 및 맛집 정보 제공 - 미수용 : 시간 없음
- [x] : 12. (제안) 예상참가비 정보 추가 제공 - 미수용 : 시간없음
- [x] : 22. (제안) 인기스토리 indicator  - 수용 ***수정완료*** 스토리상세 상단에 표기되게 함. 인기스토리에서는 필요한지 모르겠음. 가장 최신 사진이 보임
- [x] : 27. (제안) 생일 입력 시 년, 월 선택 후 일을 선택하는 것이 불편 - 미수용 : 그럼 어떻게 함?
- [x] : 30. (버그) 여정 참여 후 마이페이지에 가 보면 참가자수가 안 바뀜 - **수정완료** trips/my/[id] 라우트 핸들러 코드를 전체적으로 고침
- [x] : 31. (미구현) 찜하기 미작동 - ***구현완료*** 
- [x] : 32. (버그) 프로필 수정 후 mbti 등을 수정하고 마이페이지에 가보면 반영 안 되어 있음 - ***수정완료***
- [x] : 42. (제안) 여행지 선택에서 국내, 해외 구분 ? - 미수용 : 무슨말인지 모르겠음
- [x] : 48. (제안) 더 많은 지역 추가 - 미수용 : 시간되면 하고...
- [x] : 50. (버그) 추천 인기 버디즈 접근 -> 로그인 안내창 -> 로그인  진행 -> “이미 로그인 된 계정”이라는 경고 출력 - 재현 불가
- [x] : 61. (버그) 프로필페이지 헤더 톱니바퀴 미작동 - ***수정완료***
- [x] : 62. (버그) 팔로우 취소 alert도 success라 내용 제대로 안 읽으면 헷갈림 - 보류/미수용 - success 가 맞음
- [x] : 65. (제안) 참여하기 따닥 시 error라는 표현이 어색함 - ***수정완료*** : 디스에이블드 처리
- [x] : 68. (제안) trip 카드의 날짜가 어떤 날짜인지 명확하지 않음 (와이어프레임과 상이하네요) - ***수정완료*** - 와이어프레임과 완전히 같지는 않음
- [x] : 69. (버그) 로그인 화면에서 뒤로가기 한 번에 안 됨 - ***수정완료***
- [x] : 72. (제안) 스토리 볼 때 다음 사진으로 넘어가는 시간을 인디케이터로 표시 - 미수용 : 스토리 자동 넘어가기는 구현하지 않을 생각이었음
- [x] : 75. (제안) 검색에서 여행지 선택 시 지역 선택 후 스크롤을 위해 다른 곳 누르고 다시 스크롤 하면 다른 지역까지 중복선택된 것처럼 보임 - ***수정 완료***
- [x] : 84. (제안) recover 페이지에서 헤더 뒤로가기 클릭 시 로그인페이지를 기대했으나 홈으로 가짐 - ***수정 완료***
- [x] : 85. (제안) 이메일 회원가입 시 실시간 중복 검사 - 미수용 : 꼭 필요할까? fetch가 계속 날아가는데?
- [x] : 86. (버그) recover 시 로그인이 되어버리는 현상 - ***버그가 아님*** 원래 메일링크로 비번찾기 방식이면 로그인 되어야 하는 것임...
- [x] : 91. (제안) 소셜로그인 시 닉네임 자동 설정 - 미수용 : 소셜로그인했는데 중복이 걸렸을 경우 스무스하게 로그인 기대하는데 에러가 발생하는 상황 가능
- [x] : 95. (제안) funnel에서 뒤로가기 누르면 이전 스텝이 아니라 완전 초기화 되는 것이 불편 - 이 부분은 온보딩은 그렇지 않음(쿼리스트링으로 해결)
- [x] : 97. (버그) 여정 상세에서 다른 사람이쓴 글인데 모바일 헤더에 수정 버튼 렌더링 됨 - ***수정 완료***
- [x] : 98. (버그) 만 나이 22살, 세는 나이 24살인데 프로필에는 23살로 렌더링 - 미수용 : 이건뭐 어쩌란건지
- [x] : 100. (제안) 인스타 스토리처럼 탭스크롤로 넘기기 - ***수정완료***
- [x] : 101. (버그) 프로필 수정 시 생일에서 유효성 검사 통과 불가 - 재현 불가 : 잘 됨
- [x] : 102. (제안) 비로그인 홈 헤더 “즐거운 여정을”이 아니라 “시작해보세요”가 강조된 것이 어색 - ***수정 완료***
- [x] : 106. (제안) 부산시는 경남이 아님 - 미수용 : 귀찮으니 넘어가겠음
- [x] : 109. (제안) 홈 헤더 “0일 남았어요” 어색 - ***수정 완료*** : 메인 페이지에서도 D-0 인 친구들 다 필터링 하도록 수정
- [x] : 111. (제안) 회원 탈퇴 - 미수용 : 수파베이스에서는 너무 복잡하므로 시간 부족할 것 같아서 미수용
- [x] : 112. (버그) 프로필 진입 후 나오면 페이지 크기가 바뀜 - 재현 불가
- [x] 20. 튜토리얼에서 헤더 가리기 or 로고만 남기기 이것은 처리함
- [x] : 4. (제안) 스토리 작성 후 자동으로 넘어가지 않아 불편함 - ***수정 완료*** 넘어가게 수정
- [x] : 19. (제안) 튜토리얼에서 로고 정도는 남기더라도 헤더는 가려지게 하는 게 좋을 듯 - 수용 ***수정완료*** 가려지게 수정함

### 트러블 슈팅

- pwa 설치 유도 문구를 ios 에서도 보여지게 하기 위해 navigator.share 를 사용하였으나 되지 않는 것을 최종 확인

### 내일 계획

-

**Comment by Captain-Kim** [comment]
## 개인 기록보드 서식 (김병준)

### 오늘 한 일

- [x] 글쓰기 나이 설정 레인지 좁히기 / 미니멈 20세로 유효성까지 막고 맥시범은 알아서 낮춰보고 유효성X -> 20~70세로 레인지 범위 수정, 유효성은 20~150세로.
- [x] ‘참여하기’ 클릭 시 인원이 가득찼습니다의 경고가 error -> caution으로 수정 -> TripCard에서 에러 메시지를 라우트에서 직접 받아서 showAlert으로 전달하도록 코드가 작성되어 있는데, caution을 전달할 방법이 생각이 안 나 일단 패스.
- [x] 추천인기버디즈 나이까지 표기하기 -> 온보딩 미등록한 버디가 있어 NaN이 출력되는 경우(반환값이 string이 아닌 경우 렌더링 하지 않도록 수정)
- [x] 버디프로필에서 ‘선호하는 버디를 등록해주세요’ 문구를 ‘~ 없습니다’로 변경
- [x] 온도 지수 -> 버디즈 지수로 바꾸기
- [x] 글쓰기 여정 시작일이 오늘일 때 ‘시작 날짜가 현재 날짜보다 이전입니다’ 경고 출력 문제 해결 -> 사용자에게 착오가 있었던 듯. startDate < now로 조건문이 되어 있어 정상 작동함. -> 아님. 문제가 재현되어서 다시 살펴보니 setStartDateTimestamp(value.start.toString());  이렇게만 담고 있어서 생긴 문제였음. 이렇게 하면 시간 데이터가 담기지 않게 되고 ‘2024-08-12’ 이런 형태로만 담기는데 시간 정보를 포함하지 못해서 이런 경우에는 00:00:00초인 자정을 기준으로 날짜가 생성됨. -> 그렇다 보니 자정에서 1초라도 지나면 과거로 보게 되어 이런 현상이 생김. 따라서 시간은 따로 입력받지 않으니 임의로 23:59:59로 설정하여 어떠한 경우에도 이런 에러가 안 뜨도록 수정함.
- [x] Next.js 최적화를 위한 팀원 상호 코드리뷰 참여

### 트러블 슈팅

- 시간 관계 상 리치텍스트에디터는 개인 과제로 넘김.

### 내일 계획

- [ ] 온도지수 계산 로직 구현
- [ ] 여정 나가기 기능 구현
- [ ] 팔로워 리스트 렌더링 페이지 구현
- [ ] 만든 여정 아코디언 -> 페이지로 다시 제작
- [ ] 글쓰기 퍼널 패턴 뒤로가기 시 이전 스텝으로 가게 수정
- [ ] UT 결과보고서 PPT 제작
- [ ] 버디 프로필 글씨 크기 바꾸기
- [ ] 레디우스 값, 마진 값 손 보기
- [ ] 리치텍스트 에디터 구현
- [ ] 글쓰기 펜딩 페이지 이미지 디자이너님께 보내드리기
- [ ] 추천인기버디즈 하트로 follow 버튼 추가
- [ ] 온도지수 아이콘 넣기
- [ ] 글쓰기 아이콘 여정 리스트로 옮기기 (아이콘, 색상까지 바꾸기)
- [ ] 버디 프로필 페이지 chips 간격 4px
- [ ] 버디 프로필 수정 페이지 만들기 -> 각 내용 클릭 시 온보딩 페이지 컴포넌트 재사용 했다가 다시 돌아와서 수정 페이지에서 ‘완료’버튼 눌러야 수정되는 flow
- [ ] 버디 프로필 모바일 헤더 X 아이콘 추가
- [ ] 추천인기 버디즈 페이지 버디 선정 기준 나타내기 i 아이콘 호버 등
- [ ] 글쓰기 나이설정에서 input에 0을 못 지우는 문제 해결
